### PR TITLE
Add Basic CSS Color Parsing

### PR DESCRIPTION
**Pull request by cedrickcooke**
For now, only parses color keywords and hex notation (`#rgb`, `#rgba`, `#rrggbb`, `#rrggbbaa`). In the future we should include `rgb()` function call syntax. As such, I'm considering this only a partial implementation of #22.

Also wrote a few bonus tests since I was looking at the coverage report and they seemed fast.

### CODE REVIEW

File: color/src/commonMain/kotlin/ToColor.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package com.juul.krayon.color
+
+private const val HEX = "[a-f0-9]"
+
+private val RGB = """($HEX)($HEX)($HEX)""".toRegex(RegexOption.IGNORE_CASE)
+private val RGBA = """($HEX)($HEX)($HEX)($HEX)""".toRegex(RegexOption.IGNORE_CASE)
+private val RRGGBB = """($HEX{2})($HEX{2})($HEX{2})""".toRegex(RegexOption.IGNORE_CASE)
+private val RRGGBBAA = """($HEX{2})($HEX{2})($HEX{2})($HEX{2})""".toRegex(RegexOption.IGNORE_CASE)
```
Comment (twyatt) [inline color/src/commonMain/kotlin/ToColor.kt:8]: I suspect that regex approach is slower than processing by iterating over the hex characters.
But we can definitely optimize later (and should benchmark to verify that regex is even slower).

Created #25 (super low priority though) to potentially address this later.

**Review by twyatt — APPROVED** [review]
This is awesome! Thanks so much for integrating this so soon! ❤️

File: color/src/commonMain/kotlin/ToColor.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+public fun String.toColor(): Color =
+    keywordMap[this.toLowerCase()] ?: parseHexNotation(this)
```
Comment (twyatt) [inline color/src/commonMain/kotlin/ToColor.kt:11]: nit

Slight optimization could be to check for presence of `#` as first character (to remove hash lookup/miss) when you know you'll need to parse as hex notation.

**Review by sdonn3 — APPROVED** [review]
looking good to me

**Pull request merged**